src/main/java/swerve/ModuleIOReal.java

```diff
@@ -110,47 +111,59 @@ public void updateInputs(SwerveModuleInputs inputs) {
 
     @Override
     public double getAngle() {
-        return AngleUtil.normalize(ticksPerRad.toUnits(angleMotor.getSelectedSensorPosition()));
+        return AngleUtil.normalize(Units.rotationsToRad(angleMotor.getPosition().getValue()));
     }
 
     @Override
     public void setAngle(double angle) {
         angleSetpoint = AngleUtil.normalize(angle);
         Rotation2d error = new Rotation2d(angle).minus(new Rotation2d(currentAngle));
-        angleMotor.set(TalonFXControlMode.MotionMagic, angleMotorPosition + ticksPerRad.toTicks(error.getRadians()));
+        var motorRequest = new MotionMagicVoltage(angleMotor.getPosition().getValue() + error.getRotations());
+        angleMotor.setControl(motorRequest);
     }
 
     @Override
     public double getVelocity() {
-        return ticksPerMeter.toVelocity(driveMotor.getSelectedSensorVelocity());
+        return driveMotor.getVelocity().getValue();
     }
 
     @Override
     public void setVelocity(double velocity) {
         var angleError = new Rotation2d(angleSetpoint).minus(new Rotation2d(currentAngle));
         velocity *= angleError.getCos();
+
         driveMotorVelocitySetpoint = velocity;
-        driveMotor.set(TalonFXControlMode.Velocity, ticksPerMeter.toTicks100ms(velocity));
+        var motorRequest = new VelocityDutyCycle(
+                Units.metersPerSecondToRps(
+                        velocity,
+                        SwerveConstants.WHEEL_DIAMETER / 2
+                )
+        );
+        driveMotor.setControl(motorRequest);
```

Comment:
From the [Phoenix 6 documentation](https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/api-usage/control-requests.html#modifying-a-control-request):
> Java users should reuse control requests to prevent excessive invocation of the Garbage Collector.

You can create the request object as a class member and change the value as needed with [`withVelocity()`](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/controls/PositionVoltage.html#withVelocity(double)) (this doesn't create a new object but changes the same one).